docs/reference/technology/connectivity/networking.md

```diff
@@ -0,0 +1,119 @@
+link layer.
+
+|Stack|Data link layer|Network layer|Configurations supported|
+|-----|---------------|-------------|------------------------|
+|LwIP|Ethernet, WiFi|IPv4|DHCP, static|
+|LwIP|Ethernet, WiFi|IPv6|[RFC 4862](https://tools.ietf.org/html/rfc4862) IPv6 Stateless Address Autoconfiguration. No DHCPv6 support|
+|LwIP|PPP|IPv4,IPv6|automatic|
+|Nanostack|Ethernet|IPv6|static or [RFC 4862](https://tools.ietf.org/html/rfc4862) IPv6 Stateless Address Autoconfiguration. No DHCPv6 support|
+|Nanostack|IEEE 802.15.4|6LoWPAN|Thread or 6LoWPAN-ND+RPL|
+
+
+#### Network interfaces
+
+Network inteface is the application level API where user chooses the driver, connectivity method
+and IP stack. As each connectivity methods differ on how to configure them, these APIs are
+noninterchangeable and requires application developer to choose one. Choosing the interface, also
+automatically pulls in the network stack as a dependency.
+
+Please note that Interface API is not the same as network driver.
+Interface API is the control interface for application. Network driver implements the controlling
+API only if it requires configuration from application. From application point of view, there
+is no difference, but network driver developer needs to be aware of that.
+
+Mbed OS implements following network interface APIs
+
+* Ethernet
+* WiFi
+* Cellular (PPP)
+* 6LoWPAN-ND mesh networking
+* Thread mesh networking.
+
+Refer to [TODO: sockets/interface API reference](api/connectivity/networkinterface.md) for usage instructions.
+
+#### Network driver
+
+Network driver is a generic term to describe different APIs for connecting networking device to
+IP stack or Socket API. Each driver API has their own architecture described below.
+
+##### Ethernet driver
+
+![Emac API](emac.png)
+
+Ethernet drivers are implemented using stack independent EMAC API. As Ethernet driver requires
+no configuration, it does not implement any controlling interface for Application.
+
+##### WiFi driver
+
+![WiFi driver](wifi.png)
+
+There are two form of WiFi drivers in Mbed OS depending on which protocol layer it implements.
+WiFi drivers are either special case of Ethernet driver or they are external IP stacks. WiFi
+drivers require configuration from application, and therefore implement both, the low level EMAC API or Network stack API and high level controlling interface API called `WiFiInterface`.
+
+##### Cellular modem driver
+
+![Cellular driver](cellular.png)
+
+As with WiFi, cellular drivers have same two separate cases. If they use external IP stack,
+driver implements the Network stack API. If they use internal IP stack, LwIP, then they
+implement Serial PPP driver.
+
+##### IEEE 802.15.4 RF driver
+
+On Mesh networks, Nanostack uses IEEE 802.15.4 radios for transmitting and receiving packets.
+The RF driver implements the `NanostackRfPhy` API.
+
+This driver type has not other usecases so it is implemented in C using Nanostack specific API.
```

Comment:
Nice set of diagrams above - feels a little incomplete to be missing the diagram for this one.